### ISSUE TITLE
Unit test targets come with "Host Application" unset

### ISSUE THREAD

**Issue by maximkhatskevich** [jcampbell05/xcake#108]
even despite that definitions of the unit test targets are inside corresponding application target.

Cakefile:

``` Ruby
# https://github.com/jcampbell05/xcake
# http://www.rubydoc.info/github/jcampbell05/xcake/master/file/docs/Cakefile.md

project.name = "MyApp_iOS"
project.organization = "Maxim Khatskevich"

project.debug_configuration :Debug # for local development/debugging only, "test." environment
project.release_configuration :Staging # for alpha and beta AdHoc builds, "test." environment
project.release_configuration :RC # for release candidate AdHoc builds, "rc." environment
project.release_configuration :Production # for App Store ready AdHoc builds, "www." environment
project.release_configuration :AppStore # for App Store builds, "www." environment

#===

project.all_configurations.each do |configuration|

    configuration.settings["ENABLE_BITCODE"] = "YES"

    configuration.settings["SDKROOT"] = "iphoneos"
    configuration.settings["GCC_DYNAMIC_NO_PIC"] = "NO"
    configuration.settings["OTHER_CFLAGS"] = "$(inherited) -DNS_BLOCK_ASSERTIONS=1"
    configuration.settings["GCC_C_LANGUAGE_STANDARD"] = "gnu99"
    configuration.settings["CLANG_ENABLE_MODULES"] = "YES"
    configuration.settings["CLANG_ENABLE_OBJC_ARC"] = "YES"
    configuration.settings["ENABLE_NS_ASSERTIONS"] = "NO"
    configuration.settings["ENABLE_STRICT_OBJC_MSGSEND"] = "YES"
    configuration.settings["CLANG_WARN_EMPTY_BODY"] = "YES"
    configuration.settings["CLANG_WARN_BOOL_CONVERSION"] = "YES"
    configuration.settings["CLANG_WARN_CONSTANT_CONVERSION"] = "YES"
    configuration.settings["GCC_WARN_64_TO_32_BIT_CONVERSION"] = "YES"
    configuration.settings["CLANG_WARN_INT_CONVERSION"] = "YES"
    configuration.settings["GCC_WARN_ABOUT_RETURN_TYPE"] = "YES_ERROR"
    configuration.settings["GCC_WARN_UNINITIALIZED_AUTOS"] = "YES_AGGRESSIVE"
    configuration.settings["CLANG_WARN_UNREACHABLE_CODE"] = "YES"
    configuration.settings["GCC_WARN_UNUSED_FUNCTION"] = "YES"
    configuration.settings["GCC_WARN_UNUSED_VARIABLE"] = "YES"
    configuration.settings["CLANG_WARN_DIRECT_OBJC_ISA_USAGE"] = "YES_ERROR"
    configuration.settings["CLANG_WARN__DUPLICATE_METHOD_MATCH"] = "YES"
    configuration.settings["GCC_WARN_UNDECLARED_SELECTOR"] = "YES"
    configuration.settings["CLANG_WARN_OBJC_ROOT_CLASS"] = "YES_ERROR"
    configuration.settings["DEFINES_MODULE"] = "YES" # http://stackoverflow.com/a/27251979

    configuration.settings["CURRENT_PROJECT_VERSION"] = "1" # just default non-empty value

end

#=== Targets

application_for :ios, 9.0 do |target|

    provProfDev = ""
    provProfAdHoc = ""
    provProfAppStore = ""

    #===

    target.name = "MyApp_iOS"
    target.language = :swift

    target.all_configurations.each do |configuration|

        #=== Build Settings - Core

        configuration.product_bundle_identifier = "khatskevich.maxim.MyApp_iOS"

        configuration.settings["INFOPLIST_FILE"] = "Info/MyApp_iOS.plist"

        configuration.settings["PRODUCT_NAME"] = "$(TARGET_NAME)"
        # configuration.settings["CODE_SIGN_ENTITLEMENTS"] = "Res/MyApp-iOS.entitlements"
        # configuration.settings["GCC_PREFIX_HEADER"] = "Src/ObjC/Prefix.pch"
        # configuration.settings["FRAMEWORK_SEARCH_PATHS"] = "$(inherited)"
        # configuration.settings["LIBRARY_SEARCH_PATHS"] = "$(inherited) $(SRCROOT)/Lib/**"
        configuration.settings["ASSETCATALOG_COMPILER_APPICON_NAME"] = "MyApp_iOS"
        configuration.settings["ASSETCATALOG_COMPILER_LAUNCHIMAGE_NAME"] = "MyApp_iOS"
        # configuration.settings["SWIFT_OBJC_BRIDGING_HEADER"] = "Src/ObjC/MyApp-iOS-BridgingHeader.h"
        # configuration.settings["OTHER_LDFLAGS"] = "$(inherited) -ObjC"

        configuration.supported_devices = :ipad_only

        # make configuration name available in run time:
        # configNameFlag = "CONFIG_" + configuration.name.upcase
        # configuration.settings["GCC_PREPROCESSOR_DEFINITIONS"] = "$(inherited) " + configNameFlag + "=1" # Obj-C support
        # configuration.settings["OTHER_SWIFT_FLAGS"] = "$(inherited) -D" + configNameFlag # Swift support

        # This will show "Automatic" in Xcode,
        # relies on proper/valid "PROVISIONING_PROFILE" value:
        configuration.settings["CODE_SIGN_IDENTITY[sdk=iphoneos*]"] = nil

        #=== Build Settings - Per-configuration

        # NOTE: any configuration-specific settings will override
        # common settings that you already applied above inside this
        # "target.all_configurations.each do |configuration|" block

        if configuration.name == "Debug"

            #=== Config

            configuration.settings["PROVISIONING_PROFILE"] = provProfDev

            # configuration.settings["CODE_SIGN_ENTITLEMENTS"] = "Res/MyApp-iOS-test.entitlements"

        end

        #===

        if configuration.name == "Staging"

            #=== Config

            configuration.settings["PROVISIONING_PROFILE"] = provProfAdHoc

            # configuration.settings["CODE_SIGN_ENTITLEMENTS"] = "Res/MyApp-iOS-test.entitlements"

        end

        #===

        if configuration.name == "RC"

            #=== Config

            configuration.settings["PROVISIONING_PROFILE"] = provProfAdHoc

            # configuration.settings["CODE_SIGN_ENTITLEMENTS"] = "Res/MyApp-iOS-test.entitlements"

        end

        #===

        if configuration.name == "Production"

            #=== Config

            configuration.settings["PROVISIONING_PROFILE"] = provProfAdHoc

        end

        #===

        if configuration.name == "AppStore"

            #=== Config

            configuration.settings["PROVISIONING_PROFILE"] = provProfAppStore

        end

    end

    #=== Extra System Frameworks

    # target.system_frameworks << "AdSupport"

    #=== Source Files

    target.include_files = ["Src/**/*.*"]
    # target.include_files << "Lib/**/*.*"

    target.include_files << "Res/**/*.*"

    #=== Tests

    unit_tests_for target do |test_target|

        test_target.name = "MyApp_iOS_Tst"

        test_target.all_configurations.each do |configuration|

            configuration.settings["INFOPLIST_FILE"] = "Info/MyApp_iOS_Tst.plist"

        end

        #=== Source Files

        test_target.include_files = ["Tst/**/*.*"]

    end

    ui_tests_for target do |test_target|

        test_target.name = "MyApp_iOS_TstUI"

        test_target.all_configurations.each do |configuration|

            configuration.settings["INFOPLIST_FILE"] = "Info/MyApp_iOS_TstUI.plist"

        end

        #=== Source Files

        test_target.include_files = ["TstUI/**/*.*"]

    end

end

```

**Comment by jcampbell05** [comment]
Interesting, we do have a line that should set this here:
https://github.com/jcampbell05/xcake/blob/master/lib/xcake/project/sugar.rb#L67

Maybe the path is incorrect ?

**Comment by maximkhatskevich** [comment]
What do you mean "path is incorrect"? We do not set path to host app/target directly, right? I thought it has to be set automatically to the target that encloses unit test target definition, isn't it?

**Comment by maximkhatskevich** [comment]
Could you try to generate a project using Cakefile provided and check if host app will be set for you or not?

**Comment by maximkhatskevich** [comment]
Another details: application build schemes come not configured properly for testing at all, looks like this out of the box:

<img width="879" alt="screen shot 2016-08-11 at 11 35 44 pm" src="https://cloud.githubusercontent.com/assets/2624646/17607211/4c91a702-601c-11e6-88e1-836e9b5b37e2.png">

**Comment by chiahsien** [comment]
@jcampbell05 
Any progress for this issue? It still happens in Xcake 0.7.1 with Xcode 8.0.

**Comment by jcampbell05** [comment]
@chiahsien Looking into it, if we could create an example project with this bug then that would help me alot :)

**Comment by chiahsien** [comment]
@jcampbell05 
It happens in my project, but I can't create a sample project to reproduce it.
The `TEST_HOST` and `BUNDLE_LOADER` settings are correct, but Xcode UI isn't.
Maybe it's Xcode bug.

![settings](https://cloud.githubusercontent.com/assets/1225437/20588224/8aa00690-b24e-11e6-9fc2-f1bbb0baca86.png)
![xcode ui](https://cloud.githubusercontent.com/assets/1225437/20588225/8aa410e6-b24e-11e6-8bf8-58dec4c5e86d.png)

---

I've found another related issue.
If my target's `PRODUCT_NAME` isn't same as target's `target.name`, then I have to explicitly specify test_target's `TEST_HOST`. Otherwise the `TEST_HOST` will be set to incorrect value.

Here's the sample project.
[XcakeTest.zip](https://github.com/jcampbell05/xcake/files/610567/XcakeTest.zip)

**Comment by asifmohd** [comment]
Was going through this issue
Here is what i found

Unit test targets seem to be working fine with master xcake
Though UITest targets require something of this sort to set Target application

`
/* Begin PBXProject section */

		4B4CDB672185CCACC3107811 /* Project object */ = {
			isa = PBXProject;
			attributes = {
				LastSwiftUpdateCheck = 0830;
				LastUpgradeCheck = 0700;
				TargetAttributes = {
					208048C4315211FFA9043129 = {
						DevelopmentTeam = QM66W5D5E4;
					};
					D7353D267EE14238CCB01807 = {
						DevelopmentTeam = QM66W5D5E4;
					};
					EEEFE3F52CD14FABCBFDA965 = {
						DevelopmentTeam = QM66W5D5E4;
						TestTargetID = 208048C4315211FFA9043129;
					};
				};
			};
`

i.e they need `TestTargetID` set to the host target id for things to work

@jcampbell05 is it possible to set the `TestTargetID` somehow?

**Comment by jcampbell05** [comment]
@asifmohd I'm not sure I think the attributes for the project are easy to customize, we do that to set the Objective-C class prefix, see how we implement that :)

**Comment by jcampbell05** [comment]
Will fix this in 0.8.11, I think I need to update our target generator to add this data into the project attributes. Which I think may be possible.
